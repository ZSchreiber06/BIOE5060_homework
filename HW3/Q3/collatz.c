#include <stdio.h>

int main(int argc, char *argv[]){
  int num;
  if (argc == 1){
    printf("Format: user MUST input an integer larger than 1.\n ");
    return 1;
  }
  sscanf(argv[1], "%d", &num);

  while(num > 1){
    if(num % 2 == 0){
      num = num/2;
      printf("%d\n", num);
    }
    else{
      num = num * 3 + 1;
      printf("%d\n", num);
    }
  }

  printf("1\n");
  return 0;

}
