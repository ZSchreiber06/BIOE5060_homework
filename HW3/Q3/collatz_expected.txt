  file = fopen("collatz_expected.txt", "r")
